### PR TITLE
Multilingual anonymization

### PR DESCRIPTION
**Pull request by maks-operlejn-ds**

### CODE REVIEW

**Review by eryk-dsai — APPROVED** [review]

File: libs/experimental/langchain_experimental/data_anonymizer/presidio.py [hunk 1]
```diff
@@ -44,12 +46,23 @@
     from presidio_analyzer import EntityRecognizer, RecognizerResult
     from presidio_anonymizer.entities import EngineResult
 
+DEFAULT_LANGUAGES_CONFIG = {
+    "nlp_engine_name": "spacy",
+    "models": [
+        {"lang_code": "en", "model_name": "en_core_web_lg"},
+        # {"lang_code": "de", "model_name": "de_core_news_md"},
```
Comment (piotr-grodek-dsai) [inline libs/experimental/langchain_experimental/data_anonymizer/presidio.py:58]: Can you add a link to where one can find model names? It probably would be more helpful for people.

File: libs/experimental/langchain_experimental/data_anonymizer/presidio.py [hunk 2]
```diff
@@ -44,12 +46,23 @@
     from presidio_analyzer import EntityRecognizer, RecognizerResult
     from presidio_anonymizer.entities import EngineResult
 
+DEFAULT_LANGUAGES_CONFIG = {
+    "nlp_engine_name": "spacy",
```
Comment (piotr-grodek-dsai) [inline libs/experimental/langchain_experimental/data_anonymizer/presidio.py:55]: Is any other non-spacy model available/tested?

File: libs/experimental/langchain_experimental/data_anonymizer/faker_presidio_mapping.py [hunk 3]
```diff
@@ -27,8 +27,8 @@ def get_pseudoanonymizer_mapping(seed: Optional[int] = None) -> Dict[str, Callab
             fake.random_choices(string.ascii_lowercase + string.digits, length=26)
         ),
         "IP_ADDRESS": lambda _: fake.ipv4_public(),
-        "LOCATION": lambda _: fake.address(),
-        "DATE_TIME": lambda _: fake.iso8601(),
+        "LOCATION": lambda _: fake.city(),
```
Comment (piotr-grodek-dsai) [inline libs/experimental/langchain_experimental/data_anonymizer/faker_presidio_mapping.py:30]: nitpick but it is a sneaky way to have this changes ;) it is not related to multilngual anonymization

File: libs/experimental/langchain_experimental/data_anonymizer/presidio.py (hunk 2 above)
Comment (piotr-grodek-dsai) [reply libs/experimental/langchain_experimental/data_anonymizer/presidio.py:55]: I think at least I would put a comment about notebook

File: libs/experimental/langchain_experimental/data_anonymizer/presidio.py (hunk 1 above)
Comment (piotr-grodek-dsai) [reply libs/experimental/langchain_experimental/data_anonymizer/presidio.py:58]: I saw it is mentioned in the notebook, but I think it is nice to get information directly from code editor, without jumping to docs.

File: libs/experimental/langchain_experimental/data_anonymizer/presidio.py [hunk 4]
```diff
@@ -103,18 +124,28 @@ def add_operators(self, operators: Dict[str, OperatorConfig]) -> None:
 
 
 class PresidioAnonymizer(PresidioAnonymizerBase):
-    def _anonymize(self, text: str) -> str:
+    def _anonymize(self, text: str, language: Optional[str]) -> str:
```
Comment (mateusz-wosinski-ds) [inline libs/experimental/langchain_experimental/data_anonymizer/presidio.py]: `None` value does not tell that the english language would be used - imo better default value would be just `'en'`

**Review by piotr-grodek-dsai — APPROVED** [review]
Tiny nitpicks, but overall I like it much, very nice notebook examples.

File: libs/experimental/langchain_experimental/data_anonymizer/presidio.py (hunk 1 above)
Comment (mateusz-wosinski-ds) [inline libs/experimental/langchain_experimental/data_anonymizer/presidio.py:58]: where can we find models for other languages? There should be a comment about that

File: libs/experimental/langchain_experimental/data_anonymizer/presidio.py [hunk 5]
```diff
@@ -103,18 +124,28 @@ def add_operators(self, operators: Dict[str, OperatorConfig]) -> None:
...
         """Anonymize text.
         Each PII entity is replaced with a fake value.
         Each time fake values will be different, as they are generated randomly.
 
         Args:
             text: text to anonymize
         """
+        if language is None:
+            language = self.supported_languages[0]
```
Comment (mateusz-wosinski-ds) [inline libs/experimental/langchain_experimental/data_anonymizer/presidio.py:150]: this is not intuitive in my opinion, I would rather dismiss `None` value for `language` argument

File: libs/experimental/langchain_experimental/data_anonymizer/presidio.py (hunk 2 above)
Comment (mateusz-wosinski-ds) [inline libs/experimental/langchain_experimental/data_anonymizer/presidio.py:55]: imo comment about other engines available should be added

**Comment by mateusz-wosinski-ds** [comment]
> Tiny nitpicks, but overall I like it much, very nice notebook examples.

+1 notebook is very cool

File: libs/experimental/langchain_experimental/data_anonymizer/faker_presidio_mapping.py (hunk 3 above)
Comment (maks-operlejn-ds) [reply libs/experimental/langchain_experimental/data_anonymizer/faker_presidio_mapping.py:30]: I would leave it here, as it is comehow connected and it's sometimes difficult and time-consuming to get your PR merged by authors

File: libs/experimental/langchain_experimental/data_anonymizer/presidio.py (hunk 4 above)
Comment (maks-operlejn-ds) [reply libs/experimental/langchain_experimental/data_anonymizer/presidio.py]: What if only Spanish is provided? That's why I used:
```
if language is None:
    language = self.supported_languages[0]
```

Comment (mateusz-wosinski-ds) [reply libs/experimental/langchain_experimental/data_anonymizer/presidio.py]: then don't make a default value :p `Explicit is better than implicit`

**Pull request closed without merging**